### PR TITLE
Feature/540 bpi account subject account relation

### PR DESCRIPTION
**Pull request by skosito**
# Description

Extension of base branch PR, add many-to-many relation between bpi account and owner bpi subject accounts and modify all commands, queries and api.

## Related Issue

check base PR

## Motivation and Context

check base PR

## How Has This Been Tested

postman

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/prisma/schema.prisma [hunk 1]
```diff
@@ -23,12 +23,14 @@ model BpiSubject {
...
   nonce Int
+  ownerBpiSubjectAccounts BpiSubjectAccount[]
 }
 
 model BpiSubjectAccount {
   id                  String @id @default(uuid())
   creatorBpiSubjectId String
   ownerBpiSubjectId   String
+  bpiAccounts         BpiAccount[]
   CreatorBpiSubject   BpiSubject  @relation(fields: [creatorBpiSubjectId], references: [id], name: "creatorBpiSubject_fk")
```
Comment (ognjenkurtic) [inline examples/bri-3/prisma/schema.prisma:34]: **c**reator

File: examples/bri-3/prisma/schema.prisma [hunk 2]
```diff
@@ -23,12 +23,14 @@ model BpiSubject {
...
   OwnerBpiSubject     BpiSubject  @relation(fields: [ownerBpiSubjectId], references: [id], name: "ownerBpiSubject_fk")
```
Comment (ognjenkurtic) [inline examples/bri-3/prisma/schema.prisma:35]: **o**wner

File: examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts [hunk 3]
```diff
@@ -11,26 +13,94 @@ export class BpiAccountStorageAgent extends PrismaService {
...
+              o.OwnerBpiSubject.publicKey,
+            ),
+          );
+        }),
+      );
     });
   }
 
   async createNewBpiAccount(bpiAccount: BpiAccount): Promise<BpiAccount> {
+    const connectOwnersConnect = bpiAccount.ownerBpiSubjectAccounts.map((o) => {
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccountsStorage.agent.ts:93]: connectedOwnerBpiAccounts?

File: examples/bri-3/prisma/schema.prisma (hunk 1 above)
Comment (skosito) [reply examples/bri-3/prisma/schema.prisma:34]: dont understand this and comment below? if you are refering to small first letter that is ok, this is how is supposed to work

Comment (ognjenkurtic) [reply examples/bri-3/prisma/schema.prisma:34]: ok, was unaware this is by convention

**Review by ognjenkurtic — APPROVED** [review]

**Review by Manik-Jain — APPROVED** [review]

**Review by Kasshern — APPROVED** [review]

**Review by Therecanbeonlyone1969 — CHANGES_REQUESTED** [review]
@skosito There are requested changes to #541  that will affect this PR.

**Comment by ognjenkurtic** [comment]
copy of my comment from PR #541 

@Therecanbeonlyone1969 yes, let's please stick with the proposed models from the document @skosito outlined. As soon as CRUD milestone is reached we will start tackling all the intricate details of state storage and other MUST requirements outlined in the standard. These will be tackled in separate smaller issues to avoid tackling multiple complex topics at the same time and to enable detailed discussions in the issues before implementation even begins. Dont worry,. no MUST requirement outlined in the scope document will be left behind ;).

**Pull request merged**